### PR TITLE
[BEAM-886] Rename DoFn to OldDoFn and add deprecation message

### PR DESCRIPTION
**Pull request by sb2nov**
R: @aaltay PTAL

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by sb2nov** [comment]
R: @aaltay PTAL

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -228,6 +229,10 @@ class DoFn(WithTypeHints, HasDisplayData):
   callable object using the CallableWrapperDoFn class.
   """
 
+  def __init__(self):
+    logging.warning('Use of OldDoFn is deprecated please use DoFn instead')
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/core.py]: Use `warnings.warn`

Comment (sb2nov) [inline sdks/python/apache_beam/transforms/core.py]: Done

**Comment by aaltay** [comment]
LGTM. waiting for tests.

**Pull request closed without merging**